### ISSUE TITLE
All proton 8 versions doesnt work with AMD gpus

### ISSUE THREAD

**Issue by demanuPL** [ValveSoftware/Proton#6763]
After installing proton 8 none of games are working, also proton experimental doesnt work. I've tried all of my games and none on them even started. Problem is with AMD gpus with latest mesa. Works fine on all versions of 7 and below.

I've found this in logs:
`2858.735:0210:0214:fixme:kernelbase:AppPolicyGetProcessTerminationMethod FFFFFFFFFFFFFFFA, 000000000011FEB0
gamemodeauto: dlopen failed - libgamemode.so: cannot open shared object file: No such file or directory
gamemodeauto: dlopen failed - libgamemode.so: cannot open shared object file: No such file or directory
gamemodeauto: dlopen failed - libgamemode.so: cannot open shared object file: No such file or directory
gamemodeauto: dlopen failed - libgamemode.so: cannot open shared object file: No such file or directory
gamemodeauto: dlopen failed - libgamemode.so: cannot open shared object file: No such file or directory
gamemodeauto: dlopen failed - libgamemode.so: cannot open shared object file: No such file or directory
gamemodeauto: dlopen failed - libgamemode.so: cannot open shared object file: No such file or directory
pid 14988 != 14987, skipping destruction (fork without exec?)`

But I didnt add game mode to launch options

**Comment by kisak-valve** [comment]
Hello @demanuPL, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to some affected game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Also, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by gardotd426** [comment]
I don't think this is limited to AMD GPUs. I'm finding the same problem on Nvidia. I get a bunch of log output, the game's process shows that it's running when I run `ps -ef `, but nothing ever happens.

@kisak-valve, here's my log: https://pastebin.com/HFN8b9RU

I also want to point out:

All Proton 7 versions launch games

I've tried every Windows game I have, none of them launch with Proton 8.X. Tried GE, Experimental (both non-beta and bleeding edge), Proton GE, and Proton TKG.

**Comment by gardotd426** [comment]
..Is this not being monitored at all?

I've tried multiple kernels, system updates, driver versions (although it's obviously not the drivers since the issue appears on AMD as well).

I can at least add that up until about 4-5 days ago, Proton 8 versions DID launch games, the only for sure bug I found was that Proton versions 8 and later disable DLSS in ALL games, and DLSS works in almost all games for Proton versions up through and including 7.X. I was going to report that bug, but then Proton 8.X stopped even working at all.

**Comment by alasky17** [comment]
@demanuPL @gardotd426 Please try disabling/removing any custom programs/add-ons, especially gamemode and mangohud.

**Comment by gardotd426** [comment]
Sorry, that gist was indeed with launch options but I have tried without.

That said, I will get a gist of a Proton log WITHOUT any launch options
because I know that will help anyone looking at the logs.

I will say though, MangoHud/gamemode/gamescope/etc all work fine with
Proton 7.X and Wine 7.X/8.X in Lutris games (as does Proton 8.X). But I
still need to get you that gist, I'll have it for you when I get home

**Comment by alasky17** [comment]
When you run proton through Steam, it is using the Sniper runtime library, which has known problems with both mangohud and gamemode.

**Comment by gardotd426** [comment]
That would be a huge relief at least knowing what the problem is, although I'm confused as to why it would have worked a week or two ago with those. But yeah when I get home shortly I'll check and post a gist or if it works I'll post my results. 

What's weird is the fact that the games don't crash, they just never actually launch even though the processes are running

**Comment by gardotd426** [comment]
@alasky17 So, MangoHud is the problem as far as games not launching at all goes. Do you know if the bug is in the Sniper runtime or in MangoHu? Or could you point me toward the bug report so I could see about investigating it some more? 

Turns out gamemode is what was causing the Doom Eternal problems, I'll go report there. 

Proton 8.X still has a pretty huge bug in it, but it's a separate issue - all DLSS is disabled in any Proton version (official and non-official) from 8.0 onward. DLSS works in all games except Dying Light 2 on Proton 7.X

**Comment by ShiroBlank** [comment]
> When you run proton through Steam, it is using the Sniper runtime library, which has known problems with both mangohud and gamemode.

I've identified another program that's causing issues with proton 8, SteamVR Beta [#6782 (comment)](https://github.com/ValveSoftware/Proton/issues/6782#issuecomment-1559035800)

**Comment by alasky17** [comment]
@gardotd426 Replying here instead of the DOOM Eternal thread because that is getting pretty off topic for the actual game.  
As I understand it, gamemode tries to "be smart" and set different things for different games. So breaking only one title makes sense.  

There was a bug when Proton 8 released that caused Sniper to not always install when Proton 8 was first launched on a system.  So likely you ran into that bug, and eventually a couple of Steam Client reboots fixed the issue, at which point Sniper got correctly downloaded and you started hitting the incompatibility with MangoHud.  

I saw your comment here and on the other thread about DLSS - this hasn't been broadly reported as far as I know so it could be another problem with your system, but I have it on a list to look into.

**Comment by gardotd426** [comment]
>I saw your comment here and on the other thread about DLSS - this hasn't been broadly reported as far as I know so it could be another problem with your system, but I have it on a list to look into.

I'm not sure what could be wrong with my system, since:

 - DLSS works perfectly in **all** non-Steam games using both wine 8.x and Proton 8.x's wine builds in lutris using identical DXVK/DXVK-NVAPI/VKD3D-Proton builds. 
 - It's universally broken in Proton 8.X
 - It universally works on Proton 7.X
 - Driver version doesn't matter
 - Flatpak vs regular install doesn't matter (to clarify about all my comments, I do all my reports based on system installs, not flatpak, but I do have the flatpak for sanity-checks). 
 - I've checked to confirm that the nvngx and nvapi(64) dlls are being copied into the prefix.
 - I've deleted prefixes when moving from Proton 7.X to Proton 8.X
 - Zero launch options are being used outside of the envars `PROTON_ENABLE_NVAPI=` PROTON_HIDE_NVIDIA_GPU=0 %command%`
 - It **CANNOT** be a VKD3D-Proton error (or even a DXVK-NVAPI one) because the bug appears in Doom Eternal which is native Vulkan, as well as Wolfenstein: Youngblood which is also native Vulkan (and never had any of the other issues Doom Eternal had other than MangoHud breaking 8.X which is not an issue anymore since I've removed it)
 
This definitely seems like a Sniper Runtime issue, since that's the only variable changing between working DLSS and breaking DLSS. But anyway, I'll see if I can get some logs from a couple Steam games with `PROTON_LOG=1` and `DXVK_NVAPI_LOG_LEVEL=info` to see if DXVK_NVAPI is even being loaded in DX12 games, even though the bug appears in all titles, Vulkan or DX12. 

You're right though, I'm off-topic with this issue, so I'll post a separate one with the logs and all information I can grab when I can get it all together. Oh, and I do indeed have a separate Manjaro install and a Pop OS install on separate partitions in addition to my daily driver vanilla Arch install, with their own `~/.config` partitions, the only thing they share is $HOME and game library partitions, which will show if the issue is in Steam or not, so I'll update those installs and get everything as apples-to-apples as I can and see if the issue persists. I wouldn't be surprised if people aren't reporting this because they don't even notice, because MangoHud doesn't work, so unless they enable the fps counter in the steam overlay they won't even know they're getting lower fps in a game they already had installed, and most people don't go into their settings every time they launch a game.

**Comment by gardotd426** [comment]
DLSS is fixed, that one actually was a weird configuration issue that only
actually affected DLSS working in SOME games and no others (which is why I
didn't think it could be a config issue). It works in Dead Space Remake and
Dying Light 2  and everything else now.

**Comment by gardotd426** [comment]
Actually everything is fixed, the performance issue in Doom Eternal too, the only problem is that the solution makes zero sense. I've already explained that in the Doom Eternal thread, though, so I won't post it again here so I'm not spamming up the threads.

But @demanuPL the MangoHud folks are looking into their part in the Sniper Runtime issue right now, you said you're on AMD, can you confirm whether `MANGOHUD=1 %command%` was your problem or whether it is something else?
